lib/create-adapter.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * @packageDocumentation
+ * @hidden
+ */
+
+import { Adapter } from './common-types'
+import { RestAdapter, RestAdapterParams } from './adapters/REST/rest-adapter'
+
+type AdapterParams = {
+  cmaAdapter: Adapter
+}
+
+export type ClientParams = RestAdapterParams & AdapterParams
+
+/**
+ * @private
+ */
+export function createAdapter(params: ClientParams): Adapter {
+  if (params.cmaAdapter) {
+    return params.cmaAdapter
```

Comment:
Should be just `params.adapter`

Reply:
The parameter `adapter` is unfortunately already used by Axios adapters
https://github.com/contentful/contentful-sdk-core/blob/402769b274f61a48a1b16e408846959aaa0f74bc/src/types.ts#L44-L45

Reply:
Maybe `apiAdapter`?